### PR TITLE
cangen.c: added -c option for bursty output.

### PR DESCRIPTION
**Pull request by rasmuspeders1**
I had an idea to try and reproduce a problem I have with some bursty traffic and RX overrun erors. The bursty traffic is trying to mimic bursts of messages that might happen on CANOpen when multiple nodes sends their PDO messages as a reponse to a SYNC message.

Does this make sense to include in the cangen tool? If so here is a pull request.

Specify -c 100 to have 100 packets sent in a burst before gap.

To have a burst of 100 message with a gap of 200 milliseconds between each burst specify: -g 200 -c 100.

### CODE REVIEW

File: cangen.c [hunk 1]
```diff
@@ -149,6 +153,7 @@ int main(int argc, char **argv)
 	unsigned char verbose = 0;
 	unsigned char rtr_frame = 0;
 	int count = 0;
+	unsigned int burst_sent_count=0;
```
Comment (marckleinebudde) [inline cangen.c:156]: Please have spaces around the ` = `.

File: cangen.c [hunk 2]
```diff
@@ -166,6 +171,7 @@ int main(int argc, char **argv)
 	struct ifreq ifr;
 
 	struct timespec ts;
+	struct timespec tiny_ts;
```
Comment (marckleinebudde) [inline cangen.c:174]: What's the purpose of `tiny_ts`?

Comment (rasmuspeders1) [reply cangen.c:174]: I was having problems with the program almost immediately filling by TX buffer and throwing an error locking up. Having a very small sleep between tx messages in each burst solved the problem.

File: cangen.c (hunk 1 above)
Comment (rasmuspeders1) [reply cangen.c]: done

File: cangen.c (hunk 2 above)
Comment (marckleinebudde) [reply cangen.c:174]: Call the programm with a polltimeout `-p 1000`.

Comment (rasmuspeders1) [reply cangen.c]: ok I see. makes sense.

File: cangen.c [hunk 3]
```diff
@@ -432,7 +441,6 @@ int main(int argc, char **argv)
 			else
 				fprint_canframe(stdout, &frame, "\n", 1, maxdlen);
 		}
-
```
Comment (marckleinebudde) [inline cangen.c:435]: Unrelated change, please remove from patch,

File: cangen.c [hunk 4]
```diff
@@ -458,10 +466,11 @@ int main(int argc, char **argv)
 			fprintf(stderr, "write: incomplete CAN frame\n");
 			return 1;
 		}
+		burst_sent_count++;
+		if (gap && burst_sent_count >= burst_count) /* gap == 0 => performance test :-] */
+			if (nanosleep(&ts, NULL)) return 1;
```
Comment (marckleinebudde) [inline cangen.c:471]: Please keep the coding style, move the `return 1;` into a separate line.

File: cangen.c [hunk 5]
```diff
@@ -458,10 +466,11 @@ int main(int argc, char **argv)
...
 
-		if (gap) /* gap == 0 => performance test :-] */
-			if (nanosleep(&ts, NULL))
-				return 1;
+		if(burst_sent_count >= burst_count) burst_sent_count = 0;
```
Comment (marckleinebudde) [inline cangen.c:473]: Hmmm. Why do you reset the `burst_sent_count` to 0 here? Then the code will send in burst mode again?

Comment (rasmuspeders1) [reply cangen.c:473]: I want to have continuous bursts with gaps (if gap>0) between them.
So this way the program will send a burst, sleep for the gap time and then send another burst and so on.
if gap==0, the burst should no make a difference i think.

For example, the command "cangen can0 -c 3 -p 10 -g 1000 -L8" produces the following:
 (1588659386.791079)  can1  1E8   [8]  53 CC 70 09 25 E6 60 6C
 (1588659386.791158)  can1  375   [8]  1A 52 2E 42 0A F0 90 05
 (1588659386.791326)  can1  5CD   [8]  58 9E B3 61 10 39 BF 3B
 (1588659387.791262)  can1  3D1   [8]  79 27 26 3C 3A 15 78 5B
 (1588659387.791337)  can1  1BD   [8]  8F 65 ED 32 29 27 48 45
 (1588659387.791499)  can1  6E3   [8]  24 E6 F6 69 F1 62 07 23
 (1588659388.791451)  can1  458   [8]  32 8E 9B 1B 37 73 BC 1E
 (1588659388.791542)  can1  0FD   [8]  3E EB 7F 7C 62 A2 BE 57
 (1588659388.791667)  can1  396   [8]  7C 5D 49 5F 2D 54 CA 61
 (1588659389.791616)  can1  71B   [8]  4A 76 7D 11 86 9F 6F 3E
 (1588659389.791694)  can1  436   [8]  32 E8 5E 05 DA 6B E0 47
 (1588659389.791860)  can1  25B   [8]  A7 33 2C 01 F4 BD 0E 0A

Comment (marckleinebudde) [reply cangen.c:473]: OKay. I see.

Comment (marckleinebudde) [reply cangen.c:473]: Then, please keep the coding style, one space after `if` and the `burst_sent_count = 0;` into a new line.

File: cangen.c [hunk 6]
```diff
@@ -441,6 +441,7 @@ int main(int argc, char **argv)
 			else
 				fprint_canframe(stdout, &frame, "\n", 1, maxdlen);
 		}
+
```
Comment (marckleinebudde) [inline cangen.c]: unrelated change, please remove

**Comment by rasmuspeders1** [comment]
Basically burst mode defaults to bursts of 1 message, so gap should work as before.
